cpp/include/cudf/utilities/traits.hpp

```diff
@@ -373,6 +373,19 @@ constexpr inline bool is_floating_point(data_type type)
   return cudf::type_dispatcher(type, is_floating_point_impl{});
 }
 
+/**
+ * @brief Indicates whether `T` is a std::byte type.
+ *
+ * @param type The `data_type` to verify
+ * @return true `type` is std::byte
+ * @return false `type` is not std::byte
+ */
+template <typename T>
+constexpr inline bool is_byte()
+{
+  return std::is_same_v<std::remove_const_t<T>, std::byte>;
```

Comment:
Do we want a full decay, cv removal, reference removal, or just const removal? I’m not certain what choice is most appropriate for a trait like this.
```suggestion
  return std::is_same_v<std::decay_t<T>, std::byte>;
```

Reply:
I would even go as far as claiming something like `is_byte<volatile T&>` should fail to compile, so this seems sensible to me.

Reply:
@upsj I’m not sure if I follow. Do you mean you prefer the current state of `remove_const`? On further inquiry, the `is_integral` trait returns true for any cv-qualifiers but not references. We probably want to remove volatile qualifiers but not reference-ness. (Not that we use volatile often, but I would like to align trait behaviors with the standard where possible.)

```suggestion
  return std::is_same_v<std::remove_cv_t<T>, std::byte>;
```

Reply:
Ah sorry, that was ambiguous. Yes, I believe a trait like is_byte should be rather restrictive, or are there use cases in cuDF where you need to call it on a decltype of an expression, or a volatile type?

Reply:
I didn't remove volatile because I wasn't sure the use-case where it was needed and I would rather have it fail to compile and force a decision at that point than to assume it would be ok. I'm ok with changing it to `remove_cv_t` though.

Reply:
> @bdice: On further inquiry, the std::is_integral and std::is_floating_point traits return true for any cv-qualifiers but not references. We probably want to remove volatile qualifiers but not reference-ness.

> @upsj: I agree with staying consistent with the std library

@hyperbolic2346 I think this is consensus among reviewers that `remove_cv_t` is a good change for consistency with `std`.